### PR TITLE
Change homeworks to Wa-Tor

### PR DESCRIPTION
**Pull request by hroncok**
I want to get this approved before I go deeper to Cython and PyQt.

### CODE REVIEW

File: tutorials/05_numpy.md [hunk 1]
```diff
@@ -8,32 +8,58 @@ Výukové materiály:
...
+ * `.creatures` (stejný význam jako matice na vstupu),
+ * `.energies` (množství energie pro žraloky na daných políčkách - int)
+
+Třída bude obsahovat metodu `.tick()`, která provede uplynutí jednoho chrononu
+a pohne rybami a žraloky.
+Nejdříve se hýbou ryby, pak žraloci. Pohyby v rámci druhu probíhají postupně
+v libovolném pořadí, dejte si ale pozor, aby se některé ryby nebo žraloci
+nepohybovali v jednom chrononu vícekrát.
+
+Dále bude obsahovat metody `.count_fishes()` a `.count_sharks()`, které vrátí
```
Comment (encukou) [inline tutorials/05_numpy.md]: `count_fish`

File: tutorials/05_numpy.md [hunk 2]
```diff
@@ -8,32 +8,58 @@ Výukové materiály:
...
-* záporné honoty představují zeď (kterou nejde projít)
-* nezáporné hodnoty předstvaují průchozí prostor
-* 1 představuje cíl (ten je jen jeden, ale můžete to napsat i tak, aby jich mohlo být víc)
+[Wa-Tor]: https://en.wikipedia.org/wiki/Wa-Tor
 
-V bludišti se lze pohybovat pouze horizontálně nebo veritkálně. Hranice matice jsou neprůchozí.
+Třída v konstruktoru přijme dvourozměrnou matici ryb a žraloků, kde 0
+představuje prázdné políčko, kladná čísla představují ryby a záporná čísla
+představují žraloky. Absolutní hodnota udává, kolik chrononů uplynulo od
+poslední reprodukce daného tvora (po reprodukci se resetuje na 1).
```
Comment (encukou) [inline tutorials/05_numpy.md:25]: Volitelně by to taky melo brát matici `energies`, aby se to líp testovalo.

**Review by encukou — DISMISSED** [review]
Vypadá to dobře!

**Comment by hroncok** [comment]
Amended.

**Comment by hroncok** [comment]
Will add other Wa-Tor related homeworks here.

**Comment by hroncok** [comment]
@MarekSuchanek Please review the NumPy part. Tests and impl. in my wator repo (private, you have access).

**Comment by MarekSuchanek** [comment]
Reviewed and it (everything) looks great... 

Only thing is that not so many `numpy` features are being used. If Pandas will be about something else, I might propose to add some statistics-collector with ability to plot populations, exports and compute some interesting statistic stuff. Then it could be re-used in PyQt - some _stats_ window at the end of simulation or on pause. Or some equilibrium-settings-finder. But maybe just thinking too much about what I would like to try with it (and hopefully I will try it soon)...

**Comment by hroncok** [comment]
The Pandas lesson is designed to be before NumPy, so this would be a bit problematic. Although what you say definitely sounds interesting.

I think the amount of NumPy here is comparable with last year's task. You have a bunch of NumPy arrays and you have to traversate them and do some algorithms.

**Comment by MarekSuchanek** [comment]
Yeah, it is true that the difficulty is about the same now (`maze` had the paths, directions, distances, reachability and `wator` has the random generation, reproduction, energies and sharks/fishes movements).

Maybe we can just show it (if there will be time to implement it) as a bonus :snake:

**Comment by hroncok** [comment]
Or we can offer it as a bonus task to help people reach points if they need it (as a safety net and motivation to finish MI-PYT).

**Comment by MarekSuchanek** [comment]
Good idea :thumbsup:

**Comment by hroncok** [comment]
@MarekSuchanek Please review the Cython part. Tests and impl. in my wator repo (cython branch).

**Comment by MarekSuchanek** [comment]
Cython reviewed:

- small typo fixed in the assignment
- for implementation I would provide the `requirements.txt` (`python setup.py install` doesn't work without installed requirements beforehand) - same for the previous `numpy` task

**Comment by hroncok** [comment]
> small typo fixed in the assignment

Thanks.

> for implementation I would provide the `requirements.txt` (`python setup.py install` doesn't work without installed requirements beforehand) - same for the previous `numpy` task

Right.

**Comment by hroncok** [comment]
This is now considered finished from my POV. I'm not programming PyQt homework to tests things out just yet, we can update the homework later if we find out troubles.

**Comment by hroncok** [comment]
@MarekSuchanek Can I merge this?

**Review by MarekSuchanek — APPROVED** [review]
Yes, everything looks great!

**Pull request merged**